app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/polygon/DrawAreaTaskFragment.kt

```diff
@@ -97,6 +105,17 @@ class DrawAreaTaskFragment @Inject constructor() : AbstractTaskFragment<DrawArea
 
     // Move the camera to the last vertex, if any.
     val lastVertex = viewModel.getLastVertex() ?: return
+    moveToPosition(lastVertex)
+  }
+
+  private fun redoLastVertex() {
+    viewModel.redoLastVertex()
+
+    val lastVertex = viewModel.getLastVertex() ?: return
```

Comment:
The intention of previous comment was to also include this line into the refactored method. `maybeMoveToLastVertex()`